### DISCUSSION TITLE
Add HTML report support for license scanning

### DISCUSSION THREAD

**Discussion by dirsigler** [aquasecurity/trivy#7374] [Ideas]
### Description

It would be nice to have additional support for [License scanning](https://aquasecurity.github.io/trivy/v0.54/docs/scanner/license/) as an HTML page via the template format.
For now the [template](https://github.com/aquasecurity/trivy/blob/main/contrib/html.tpl) only supports misconfigurations and vulnerabilities.

### Target

None

### Scanner

License

**Comment by afdesk** [comment]
there is a Trivy plugin for it
```sh
$ trivy plugin install scan2htm
```

**Reply by fatihtokus** [reply]
Thanks, @afdesk . trivy plugin install scan2html

I am the maintainer of the plugin, have you used it @dirsigler  ?

**Reply by dirsigler** [reply]
I did not, but honestly it feels kinda weird to have the template already there in the "contrib" folder, but then it is not up to date with the support of different scanners.

So technically the plugin might solve my issue, but I think the contrib HTML template could also deserve some love, else it can be skipped entirely and people just use plugins.

**Reply by itaysk** [reply]
> else it can be skipped entirely and people just use plugins.

that's the plan. the contrib templates will be removed at some point in favor of community maintained plugins

**Reply by fatihtokus** [reply]
I agree with @itaysk . trivy is a great scanner with some basic reporting functionalities. Rather than making it a giant monolith tool, introducing pluggable enhanced features via plugins will boost the community.

**Reply by jaudriga** [reply]
AFAICS https://github.com/fatihtokus/scan2html does not support visualizing licenses as of today.

At least I did not find a way to do it and if I see that correctly there is also an open issue for it https://github.com/fatihtokus/scan2html/issues/37 .

If that is not correct, can you please be more precise on how to use this plugin to visualize licenses?

**Reply by fatihtokus** [reply]
Hi @jaudriga, 

We are working on that, it will be released in the next release. Please bear with us.

**Reply by fatihtokus** [reply]
Hi @jaudriga ,

We have released the feature. Please have a look and share your feedback.

trivy scan2html image --scanners license alpine:latest --scan2html-flags --output interactive_report.html

<img width="1486" alt="image" src="https://github.com/user-attachments/assets/9e140520-359a-4993-ad0e-467e09ba20ca" />

**Reply by jaudriga** [reply]
Looks good. I am trying it out for an npm project with `trivy scan2html fs --scanners license . --scan2html-flags --help --output interactive_report.html`. It seems to list all relevant dependencies and produce a nice overview.

There are two small things that I noticed:

* optional npm dependencies do not seem to be supported and will be left out (this seems to be a general trivy limitation, see https://github.com/aquasecurity/trivy/discussions/7165 )
* link appears to be empty for each dependency

**Reply by fatihtokus** [reply]
I am glad to hear that.

- Yes, it is not supported by trivy yet.
- Yeah, I noticed that as well. Unfortunately, trivy is returning empty for 'Link', maybe @afdesk and @itaysk  can help with that
 {
          "Severity": "LOW",
          "Category": "notice",
          "PkgName": "libretls",
          "FilePath": "",
          "Name": "MIT",
          "Text": "",
          "Confidence": 1,
          "Link": ""   <---- THIS is EMPTY
  }
